src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,137 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    // Fields
+    private static float LOAD_FACTOR = 0.75f;
+    private static int INITIAL_CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
 
+    // Static class
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private int hash;
+        private Node<K, V> next;
+
+        Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+            this.hash = key != null ? key.hashCode() : 0;
+        }
+    }
+
+    // Constructors
+    public MyHashMap() {
+        table = (Node<K, V>[])new Node[INITIAL_CAPACITY];
+        threshold = (int)(table.length * LOAD_FACTOR);
+    }
+
+    // Public methods
     @Override
     public void put(K key, V value) {
-
+        if (++size > threshold) {
+            table = resize();
+        }
+        putToNewTable(new Node<K, V>(key, value), table);
+        threshold = (int)(table.length * LOAD_FACTOR);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key, table.length);
+        if (table[index] != null) {
+            Node<K, V> foundNode = searchInBacket(key, index);
+            return foundNode != null ? foundNode.value : null;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
     }
+
+    // Private methods
+    private Node<K,V> searchInBacket(K key, int index) {
+        Node<K,V> current = table[index];
+        while (current != null) {
+            if (current.key == key
+                    || key != null && key.equals(current.key)) {
+                return current;
```

Comment:
```suggestion
                return current.value;
```
можна тут одразу повернути значення і не потрібно буде писати тернарник в getValue